interface/fax/fax_view.php

```diff
@@ -35,41 +39,84 @@
     }
 
     while (!feof($jfhandle)) {
-        $line = trim(fgets($jfhandle));
-        if (substr($line, 0, 12) == '!postscript:') {
-            $ffname = $GLOBALS['hylafax_basedir'] . '/' .
-                substr($line, strrpos($line, ':') + 1);
-            break;
-        }
-    }
+	    $line = trim(fgets($jfhandle));
+
+	    if (strstr($line, '!')) {
+		    $docs[] = $GLOBALS['hylafax_basedir'] . '/' . substr($line, strrpos($line, ":")+1);
+	    }
 
+    }
     fclose($jfhandle);
-    if (!$ffname) {
-        die(xlt("Cannot find postscript document reference in ") . text($jfname));
+
+    $tmp1 = array();
+    $tmp2 = 0;
+    $inames = '';
+    $faxcache = $GLOBALS['OE_SITE_DIR'] . "/faxcache/tmp";
+    if (!is_dir($faxcache)) {
+	    exec("mkdir -p " . escapeshellarg($faxcache));
+    }
+    foreach ($docs as $filepath) {
+	    $filebase = basename($filepath) . ".pdf";
+	    if (preg_match('/ps/', $filepath)) {
+		    $tmp0 = exec("ps2pdf " . escapeshellarg($filepath) . " " . escapeshellarg($faxcache) . "/" . escapeshellarg($filebase), $tmp1, $tmp2);
+		    if ($tmp2) {
+			    die(xlt("ps2pdf returned $tmp2: $tmp0 "));
+		    }
+	    }
+	    else if (preg_match('/tif/', $filepath)) {
+		    $tmp0 = exec("tiff2pdf -p letter -o " . escapeshellarg($faxcache) . "/" . escapeshellarg($filebase) . " " . escapeshellarg($filepath), $tmp1, $tmp2);
+		    if ($tmp2) {
+			    die(xlt("tiff2pdf returned $tmp2: $tmp0 "));
+		    }
+	    }
+	    else {
+		    $tmp0 = exec("cp " . escapeshellarg($filepath) . " " . escapeshellarg($faxcache) . "/" . escapeshellarg($filebase), $tmp1, $tmp2);
+		    if ($tmp2) {
+			    die(xlt("cp returned $tmp2: $tmp0 "));
+		    }
+	    }
+	    $inames .= " " . $faxcache . "/" . $filebase;
+    }
+
+    if (count($docs) > 1) {
+	    $ffname = $faxcache . "/combine.pdf";
+	    $tmp0 = exec("gs -dNOPAUSE -sDEVICE=pdfwrite -sOUTPUTFILE=" . escapeshellarg($ffname) . " -dBATCH" . $inames, $tmp1, $tmp2);
+	    if ($tmp2) {
+		    die(xlt("gs returned $tmp2: $tmp0 "));
+	    }
+    }
+    else {
+	    $ffname = $filepath;
+    }
+
+    if (!file_exists($jfname)) {
+        die(xlt("Cannot find document reference in ") . text($jfname));
```

Comment:
this is a good one to keep as xlt, but need to remove the trailing whitespace in the constant

Reply:
ok